src/integrations/buckaroo/class-buckaroo.php

```diff
@@ -0,0 +1,571 @@
+   * @param string $redirect_url_error  URL to redirect on error.
+   * @param string $redirect_url_reject URL to redirect on reject.
+   * @return void
+   */
+  public function set_redirect_urls( string $redirect_url, string $redirect_url_cancel, string $redirect_url_error, string $redirect_url_reject ) {
+    $this->set_return_url( $redirect_url );
+    $this->set_return_url_cancel( $redirect_url_cancel );
+    $this->set_return_url_error( $redirect_url_error );
+    $this->set_return_url_reject( $redirect_url_reject );
+  }
+
+  /**
+   * Generates the invoice name based on submitted data + salted with time (meaning it should always be unique)-
+   *
+   * @param  array $params Parameters from request.
+   * @return string
+   */
+  public function generate_debtor_reference( array $params ) {
+    $prefix      = 'debtor';
+    $data_hash   = hash( 'crc32', wp_json_encode( $params ) );
+    $random_hash = hash( 'crc32', uniqid() );
+    return "{$prefix}-{$data_hash}-{$random_hash}";
+  }
+
+  /**
+   * Generates the invoice name based on submitted data + salted with time (meaning it should always be unique)-
+   *
+   * @param  array $params Parameters from request.
+   * @return string
+   */
+  public function generate_invoice_name( array $params ) {
+    $prefix      = 'invoice';
+    $data_hash   = hash( 'crc32', wp_json_encode( $params ) );
+    $random_hash = hash( 'crc32', uniqid() );
+    return "{$prefix}-{$data_hash}-{$random_hash}";
+  }
+
+  /**
+   * Generates the invoice name based on submitted data + salted with time (meaning it should always be unique)-
+   *
+   * @param  array $params Parameters from request.
+   * @return string
+   */
+  public function generate_purchase_id( array $params ) {
+    $prefix      = 'purchase-id';
+    $data_hash   = hash( 'crc32', wp_json_encode( $params ) );
+    $random_hash = hash( 'crc32', uniqid() );
+    return "{$prefix}-{$data_hash}-{$random_hash}";
+  }
+
+  /**
+   * Generates the correct authorization header.
+   *
+   * @param array  $post_array   Array of post data we're sending to Buckaroo.
+   * @param string $buckaroo_uri Buckaroo URI we're posting to.
+   * @return string
+   */
+  protected function generate_authorization_header( array $post_array, string $buckaroo_uri ): string {
+    $this->verify_buckaroo_info_exists();
+    $website_key = \apply_filters( Filters::BUCKAROO, 'website_key' );
+    $secret_key  = \apply_filters( Filters::BUCKAROO, 'secret_key' );
+    $post        = \wp_json_encode( $post_array );
+    $md5         = md5( $post, true );
+    $post        = base64_encode( $md5 ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+    $uri         = strtolower( rawurlencode( $buckaroo_uri ) );
+    $nonce       = \wp_rand( 0000000, 9999999 );
+    $time        = time();
+
+    $hmac = $website_key . 'POST' . $uri . $time . $nonce . $post;
+    $s    = hash_hmac( 'sha256', $hmac, $secret_key, true );
```

Comment:
`$s` is a really bad variable name.

Reply:
Good point, I copied the Buckaroo's authorization example and forgot to change this after.